ast/src/parsed/asm.rs

```diff
@@ -343,6 +348,22 @@ pub struct Machine<T> {
     pub statements: Vec<MachineStatement<T>>,
 }
 
+impl<T: Clone> Machine<T> {
+    /// Returns a vector of all local variables / names defined in the machine.
+    pub fn local_names(&self) -> Box<dyn Iterator<Item = &String> + '_> {
```

Comment:
why not just `Vec<String>`?

Reply:
I had `Vec<&String>` before. Mainly wanted to do it this way, because it is more rusty and also wanted to check if it will be complicated. Also it saves heap allocations, but not sure if this is a strong argument.

Reply:
ok